internal/storage/storage.go

```diff
@@ -0,0 +1,382 @@
+		statefulSet := new(appsv1.StatefulSet)
+
+		err := converter.FromUnstructured(s.Unstructured.UnstructuredContent(), statefulSet)
+		if err != nil {
+			return false, fmt.Errorf("convert Unstructured to StatefulSet failed: %w", err)
+		}
+
+		return statefulSet.Spec.Template.Spec.HostNetwork, nil
+	case podString:
+		pod := new(v1.Pod)
+
+		err := converter.FromUnstructured(s.Unstructured.UnstructuredContent(), pod)
+		if err != nil {
+			return false, fmt.Errorf("convert Unstructured to Pod failed: %w", err)
+		}
+
+		return pod.Spec.HostNetwork, nil
+	case jobString:
+		job := new(batchv1.Job)
+
+		err := converter.FromUnstructured(s.Unstructured.UnstructuredContent(), job)
+		if err != nil {
+			return false, fmt.Errorf("convert Unstructured to Job failed: %w", err)
+		}
+
+		return job.Spec.Template.Spec.HostNetwork, nil
+	case cronJobString:
+		cronJob := new(batchv1.CronJob)
+
+		err := converter.FromUnstructured(s.Unstructured.UnstructuredContent(), cronJob)
+		if err != nil {
+			return false, fmt.Errorf("convert Unstructured to CronJob failed: %w", err)
+		}
+
+		return cronJob.Spec.JobTemplate.Spec.Template.Spec.HostNetwork, nil
+	}
+	return false, nil
+}
+
+func (s *StoreObject) ShortPath() string {
+	return strings.Join(strings.Split(s.Path, string(os.PathSeparator))[1:], string(os.PathSeparator))
+}
+
+func (s *StoreObject) Identity() string {
+	kind := s.Unstructured.GetKind()
+	name := s.Unstructured.GetName()
+	namespace := s.Unstructured.GetNamespace()
+
+	if namespace == "" {
+		return fmt.Sprintf("kind = %s ; name = %s", kind, name)
+	}
+	return fmt.Sprintf("kind = %s ; name = %s ; namespace = %s", kind, name, namespace)
+}
+
+type UnstructuredObjectStore struct {
+	Storage map[ResourceIndex]StoreObject
+}
+
+func NewUnstructuredObjectStore() *UnstructuredObjectStore {
+	return &UnstructuredObjectStore{Storage: make(map[ResourceIndex]StoreObject)}
+}
+
+func (s *UnstructuredObjectStore) Put(path string, object map[string]any, raw []byte) error {
+	var u unstructured.Unstructured
+	u.SetUnstructuredContent(object)
+
+	storeObject := StoreObject{Path: path, Unstructured: u, Hash: NewSHA256(raw)}
+
+	index := GetResourceIndex(storeObject)
+	if _, ok := s.Storage[index]; ok {
+		// for cert-manager migration we have duplicated resources for legacy version
+		// it's ok for cluster but is not expected by tests. Remove it after legacy version will be removed
+		if strings.Contains(index.AsString(), "ClusterIssuer") || strings.HasPrefix(index.AsString(), "d8-cert-manager") {
+			return nil
+		}
+		return fmt.Errorf("object %q already exists in the object store", index.AsString())
+	}
+
+	s.Storage[index] = storeObject
+	return nil
+}
+
+func (s *UnstructuredObjectStore) Get(key ResourceIndex) StoreObject {
+	return s.Storage[key]
+}
+
+func (s *UnstructuredObjectStore) Exists(key ResourceIndex) bool {
+	_, ok := s.Storage[key]
+	return ok
+}
+
+func (s *UnstructuredObjectStore) Close() {
+	for k := range s.Storage {
```

Comment:
looks like it will be faster
```suggestion
	s.Storage = make(map[ResourceIndex]StoreObject)
```

Reply:
thx!